src/dxtbx/model/boost_python/beam.cc

```diff
@@ -221,6 +224,101 @@ namespace dxtbx { namespace model { namespace boost_python {
     return b;
   }
 
+  // PolyBeam
+
+  std::string polybeam_to_string(const PolyBeam &beam) {
+    std::stringstream ss;
+    ss << beam;
+    return ss.str();
+  }
+
+  struct PolyBeamPickleSuite : boost::python::pickle_suite {
+    static boost::python::tuple getinitargs(const PolyBeam &obj) {
+      return boost::python::make_tuple(obj.get_sample_to_source_direction(),
+                                       obj.get_divergence(),
+                                       obj.get_sigma_divergence(),
+                                       obj.get_polarization_normal(),
+                                       obj.get_polarization_fraction(),
+                                       obj.get_flux(),
+                                       obj.get_transmission());
+    }
+  };
+
+  static PolyBeam *make_polybeam(vec3<double> direction) {
+    return new PolyBeam(direction);
+  }
+
+  static PolyBeam *make_polybeam_w_divergence(vec3<double> direction,
+                                              double divergence,
+                                              double sigma_divergence,
+                                              bool deg) {
+    PolyBeam *beam = NULL;
+    if (deg) {
+      beam =
+        new PolyBeam(direction, deg_as_rad(divergence), deg_as_rad(sigma_divergence));
+    } else {
+      beam = new PolyBeam(direction, divergence, sigma_divergence);
+    }
+    return beam;
+  }
+
+  static PolyBeam *make_polybeam_w_all(vec3<double> direction,
+                                       double divergence,
+                                       double sigma_divergence,
+                                       vec3<double> polarization_normal,
+                                       double polarization_fraction,
+                                       double flux,
+                                       double transmission,
+                                       bool deg) {
+    PolyBeam *beam = NULL;
+    if (deg) {
+      beam = new PolyBeam(direction,
+                          deg_as_rad(divergence),
+                          deg_as_rad(sigma_divergence),
+                          polarization_normal,
+                          polarization_fraction,
+                          flux,
+                          transmission);
+    } else {
+      beam = new PolyBeam(direction,
+                          divergence,
+                          sigma_divergence,
+                          polarization_normal,
+                          polarization_fraction,
+                          flux,
+                          transmission);
+    }
+    return beam;
+  }
+
+  template <>
+  boost::python::dict to_dict<PolyBeam>(const PolyBeam &obj) {
+    boost::python::dict result;
+    result["__id__"] = "polychromatic";
+    result["direction"] = obj.get_sample_to_source_direction();
```

Comment:
Here we are serializing the direction as sample-to-source, i.e. `-s0.normalize()`. I do think this is the way dxtbx keeps beam direction, which has always seemed a little perverse, but anyway that's the convention.